BUILD.md

```diff
@@ -0,0 +1,212 @@
+  $ export https_proxy=http://example-proxy-server.com:3129
+  $ export no_proxy=example.com,another-example.com,127.0.0.0
+```
+
+**Note:** You will want to add these to either your .profile or .bashrc file
+so you don't have to reset them everytime you open a new shell.
+
+### Docker Proxy Settings
+
+To configure Docker to work with an HTTP or HTTPS proxy server, follow the
+instructions for your  operating system:
+
+* macOS - See the instructions for proxy configuration in
+[Get Started with Docker for Mac](https://docs.docker.com/docker-for-mac/)
+
+* Linux - See the instructions for proxy configuration in
+[Control and configure Docker with Systemd](https://docs.docker.com/engine/admin/systemd/#httphttps-proxy)
+
+Create a file `/etc/systemd/system/docker.service.d/http-proxy.conf` with the
+contents:
+
+**Warning:** The example URLs and port numbers used below are examples only.
+Please substitute the actual URL, with actual port numbers, used in your
+environment. Contact your network administrator for the information if
+necessary.
+
+```text
+[Service]
+Environment="HTTP_PROXY=http://example-proxy-server.com:3128" "HTTPS_PROXY=http://example-proxy-server.com:3129" "http_proxy=http://example-proxy-server.com:3128" "https_proxy=http://example-proxy-server.com:3129" "no_proxy=example.com,another-example.com,127.0.0.0"
+```
+
+Restart docker
+```bash
+$ sudo systemctl daemon-reload
+$ sudo systemctl restart docker
+```
+
+Verify that the configuration has been loaded:
+```bash
+$ systemctl show --property=Environment docker
+Environment=HTTP_PROXY=http://proxy.example.com:80/
+```
+
+Step Two: Clone The Repository
+-------------
+
+You'll need to have git installed in order to clone the Sawtooth source
+code repository. You can find up-to-date installation instructions
+at [Getting Started - Installing Git](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git).
+
+Open up a terminal and run the following:
+
+```bash
+   $ cd $HOME
+   $ mkdir sawtooth
+   $ cd sawtooth
+   $ git clone https://github.com/hyperledger/sawtooth-core.git
+```
+
+Step Three: Initializing Your Environment
+-------------
+
+Some of the components of Sawtooth depend partially on code that must first be
+built. This includes generating protobuf classes for each language. To build
+all of the components of Sawtooth, run this command:
+
+```bash
+  $ bin/build_all
+```
+
+By default this builds all the components for all the components and sdks in
+the repository. This is a very slow process. It is recommended that you only
+build for the language dependencies you need for your development purposes. The
+minimum requirements for running the Sawtooth validator is the python build:
+
+```bash
+  $ bin/build_all -l python
+```
+
+To get details on all the options for the build_all script
+
+```bash
+  $ bin/build_all -h
+```
+
+If you are working on the core validate the ‘python’ language is all that is
```

Comment:
s/validate/validator; comma after "validator"; change the end of the sentence to "only the 'python' language is required.